osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModAlternate : ModAlternate<OsuHitObject, OsuAction>
+    {
+        private OsuAction? lastActionPressed;
+        private OsuAction? lastActionReleased;
+
+        protected override void OnBreakEnd()
+        {
+            lastActionPressed = lastActionReleased = null;
+        }
+
+        protected override bool OnPressed(OsuAction action)
+        {
+            if (lastActionPressed != null && lastActionPressed == action)
```

Comment:
The null-check is redundant. The incoming `OsuAction` is of a value type, so it can't ever be equal to `null`.

Reply:
`lastActionPressed` and `lastActionReleased` are nullables. They are both set to null at the start of the map and after breaks.

Reply:
They are, but what I'm getting at is that `lastActionPressed == action` can't ever be true if `lastActionPressed` is null.